src/main/java/org/opensearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -226,17 +224,21 @@ public KNNVectorFieldMapper build(BuilderContext context) {
                         knnMethodContext);
             }
 
-            ModelContext modelContext = this.modelContext.get();
-            if (modelContext != null) {
+            String modelId = this.modelId.get();
```

Comment:
May be different name to avoid collision with instance name?

Reply:
Sure, I can update.